### PR TITLE
Place XLAShard data on CPU on creation

### PR DESCRIPTION
**Pull request by jonb377**
As a follow-up from #5016, we should place the shards on CPU by default. This makes it clear to the user that the shards will not remain up-to-date with the underlying XLAShardedTensor, and it eases interoperability of the shards' data with other tensors.

Expanding on the second point, the shards are backed by PjRtData when returned from `_get_local_shards`. In SPMD mode, we expect all computation inputs to have PjRtShardedData handles, which makes the on-device shards incompatible with SPMD execution. In order to use the shards in an on-device computation, they will now need to be transferred back to the device, which implies implicit replication.

### CODE REVIEW

File: test/spmd/test_xla_sharding.py [hunk 1]
```diff
@@ -48,11 +48,11 @@ def test_xla_shards(self):
       self.assertEqual(shard.data.shape, (shard_len,))
       start, end = (i, i + 1) * shard_len
       expected = torch.arange(start, end, dtype=torch.float32)
-      self.assertTrue(torch.allclose(shard.data.cpu(), expected))
+      self.assertTrue(torch.allclose(shard.data, expected))
```
Comment (JackCaoG) [inline test/spmd/test_xla_sharding.py:51]: can we add a assert above to check that `shard.data` is on cpu device. Just to make check more explict.

Comment (jonb377) [reply test/spmd/test_xla_sharding.py:52]: Good point, added

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Review by JackCaoG — APPROVED** [review]

File: torch_xla/experimental/xla_sharded_tensor.py [hunk 2]
```diff
@@ -97,7 +94,7 @@ def __new__(cls, elem: torch.Tensor, *args, **kwargs):
 
   # Shards on the devices are materialized/available after the lazy
   # execution of the SPMDPartitioned HLO graph. Each XLAShard points
-  # to torch.Tensor (xla::device_data). The shards represent a snapshot
+  # to torch.Tensor (xla::device_data). The shards represent a snapshot on CPU
```
Comment (yeounoh) [inline torch_xla/experimental/xla_sharded_tensor.py:97]: nit. Could we update the comment here, "execution of the partitioned HLO graph. Each XLAShard points to torch.Tensor. The shards represent a snapshot on CPU, detached from the global tensor." ?

**Review by yeounoh — APPROVED** [review]
Left a minor comment, LGTM.

**Pull request merged**